### DISCUSSION TITLE
Upgrading Mirth Connect to 4.5.0

### DISCUSSION THREAD

**Discussion by elemle** [nextgenhealthcare/connect#6156] [Q&A] [answered]
After upgrading Mirth Connect Version from 4.4.2  to  4.5.0.  
(I am hoping for a quick fix, instead of spending hours/days reading through upgrade info and log statements.)

I tried to connect to the administrator (to get the logon popup window) and I get the following error message.

<html>
<body>
<h2 style="color: rgb(0, 0, 0); font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">HTTP ERROR 500 java.lang.UnsupportedOperationException: Use .getMatched(String) instead</h2>
URI: | /webstart
-- | --
500
java.lang.UnsupportedOperationException: Use .getMatched(String) instead
-
java.lang.UnsupportedOperationException: Use .getMatched(String) instead

<h3 style="color: rgb(0, 0, 0); font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Caused by:</h3><pre style="color: rgb(0, 0, 0); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">java.lang.UnsupportedOperationException: Use .getMatched(String) instead</pre>
</body>
</html>

Mirth Connect Server 4.4.2
Built on October 23, 2023
Java version: 1.8.0_372

**Comment by ab-mg-23** [comment, marked as answer]
How are you applying the upgrade? There have been reports when using the installer sometimes older JARs of dependencies get left behind post install and it causes classpath issues amongst other oddities. Jetty was upgraded between 4.4.2 (Jetty 9.4.44) and 4.5.0 (Jetty 9.4.53). Starting in jetty 9.4.47 `PathMapping.getMatch(String)` was deprecated and throws an `UnsupportedException` so my suspicion is there's a mix of Jetty JARs still in `server-lib` after your uprgade.

**Reply by elemle** [reply]
Yes. Used the .sh installer.
Okay, thanks.  Will look into this today.

**Reply by elemle** [reply]
That FIXED it!  I removed all the 9.4.44 jar files etc.   Now onto the next problem of not being able to login.
